src/main/java/janet/Janet.java

```diff
@@ -0,0 +1,126 @@
+
+    /**
+     * Class constructor for Janet.
+     */
+    public Janet() {
+        storage = new Storage("janet.txt");
+        try {
+            tasks = storage.readSave();
+        } catch (IOException | ClassNotFoundException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Returns a command that represents the user's input in the chat box.
+     *
+     * @param input String entered by the user into the chat box
+     * @return Command that Janet understands
+     */
+    Command getCommandFromInput(String input) {
+        try {
+            return Parser.parseUserInput(input);
+        } catch (JanetException e) {
+            return new Command("echo", e.toString());
+        }
+    }
+
+    /**
+     * Returns the string to be shown in the chat as Janet's reply to user commands.
+     *
+     * @param command Command representing user's input
+     * @return String representing Janet's reply
+     * @throws IOException When unable to write the save
+     */
+    String getResponseFromCommand(Command command) throws IOException {
+        assert(command.getOperation() != null);
+        switch (command.getOperation()) {
+        case "bye":
+            Platform.exit();
+            return "";
+        case "list":
+            return Ui.taskListString(tasks);
+        case "done":
+            return handleTaskDone(command);
+        case "todo":
+            return handleNewToDo(command);
+        case "deadline":
+            return handleNewDeadline(command);
+        case "event":
+            return handleNewEvent(command);
+        case "delete":
+            return handleDelete(command);
+        case "find":
+            return handleFind(command);
+        case "schedule":
+            return handleSchedule(command);
+        case "echo":
+            return command.getDescription();
+        default:
+            return "";
+        }
+    }
+
+    private String handleTaskDone(Command command) throws IOException {
+        Task task = tasks.get(command.getIndex() - 1);
+        task.setDone(true);
+        storage.writeSave(tasks);
+        return Ui.doneString(task);
+    }
+
+    private String handleNewToDo(Command command) throws IOException {
+        ToDo task = new ToDo(command.getDescription());
+        tasks.add(task);
+        storage.writeSave(tasks);
+        return Ui.addedString(task, tasks.size());
+    }
+
+    private String handleNewDeadline(Command command) throws IOException {
+        Deadline task = new Deadline(command.getDescription(), command.getTime());
+        tasks.add(task);
+        storage.writeSave(tasks);
+        return Ui.addedString(task, tasks.size());
+    }
+
+    private String handleNewEvent(Command command) throws IOException {
+        Event task = new Event(command.getDescription(), command.getTime());
+        tasks.add(task);
+        storage.writeSave(tasks);
+        return Ui.addedString(task, tasks.size());
+    }
```

Comment:
Would this be better if it was all in a single method and handles the different cases? Most of the code are repeated code so it might help to make it shorter. Other than that, looks good to me.

Reply:
Thanks for your feedback. I was considering doing this but realised I would have many if-else statements. Now that I think about it further, it may be possible to use polymorphism to achieve this 😄